packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1273 @@
+      );
+
+      if (!actualAgreementDocument) {
+        fail(`impossible to decode ${expectedStoreEvent.type} data`);
+      }
+      return actualAgreementDocument;
+    };
+
+    async function getExpectedAgreementEventV1<T>(
+      agreementId: AgreementId | undefined,
+      version: number | undefined,
+      expectedStoreEvent: Omit<StoredEvent, "data" | "event_version">,
+      decoderFunction: (
+        event: StoredEvent,
+        expectedStoreEvent: Omit<StoredEvent, "data" | "event_version">
+      ) => T
+    ): Promise<T> {
+      expect(agreementId).toBeDefined();
+      if (!agreementId) {
+        fail(
+          `Unhandled error: returned agreementId is undefined for ${agreementId}`
+        );
+      }
+
+      const actualAgreementData: StoredEvent | undefined =
+        version !== undefined
+          ? await readAgreementEventByVersion(agreementId, version, postgresDB)
+          : await readLastAgreementEvent(agreementId, postgresDB);
+
+      if (!actualAgreementData) {
+        fail("Creation fails: agreement not found in event-store");
+      }
+
+      expect(actualAgreementData).toMatchObject({
+        ...expectedStoreEvent,
+        event_version: 1,
+      });
+
+      return decoderFunction(actualAgreementData, expectedStoreEvent);
+    }
+
+    function toAgreementStampsV1(agreement: Agreement): StampsV1 {
+      const stamps: StampsV1 = {};
+
+      if (agreement.stamps.submission) {
+        stamps.submission = toStampV1(agreement.stamps.submission);
+      }
+
+      if (agreement.stamps.activation) {
+        stamps.activation = toStampV1(agreement.stamps.activation);
+      }
+
+      if (agreement.stamps.rejection) {
+        stamps.rejection = toStampV1(agreement.stamps.rejection);
+      }
+
+      if (agreement.stamps.suspensionByProducer) {
+        stamps.suspensionByProducer = toStampV1(
+          agreement.stamps.suspensionByProducer
+        );
+      }
+
+      if (agreement.stamps.suspensionByConsumer) {
+        stamps.suspensionByConsumer = toStampV1(
+          agreement.stamps.suspensionByConsumer
+        );
+      }
+
+      if (agreement.stamps.upgrade) {
+        stamps.upgrade = toStampV1(agreement.stamps.upgrade);
+      }
+
+      if (agreement.stamps.archiving) {
+        stamps.archiving = toStampV1(agreement.stamps.archiving);
+      }
+
+      return stamps;
+    }
+
+    async function uploadDocument(path: string, name: string): Promise<void> {
+      await fileManager.storeBytes(
+        config.s3Bucket,
+        config.consumerDocumentsPath,
+        path,
+        name,
+        Buffer.from("large-document-file")
+      );
+      expect(await fileManager.listFiles(config.s3Bucket)).toContainEqual(
+        `${config.consumerDocumentsPath}/${path}/${name}`
+      );
+    }
+
+    it("should succeed with valid Verified and Declared attributes", async () => {
+      const authData = {
+        ...getRandomAuthData(),
+        userRoles: [userRoles.INTERNAL_ROLE],
```

Comment:
Upgrade agreement should be authorized only for `ADMIN_ROLE`, not for internal - also, internal role skips some of the assertions on the organizationId so this test is actually skipping some assertions that should be required.
See https://github.com/pagopa/interop-be-monorepo/blob/b5e050dc9c3e790532ab3816cd04848c0eeccdb9/packages/agreement-process/src/model/domain/validators.ts#L134

If you remove this override, `getRandomAuthData` should return already auth data with role `admin`

Reply:
By the way, having assertions on roles inside the logic methods would have avoided this issue -- which is more or less what this PR does if I also integrate the comment from @AsterITA: https://github.com/pagopa/interop-be-monorepo/pull/314#discussion_r1549111254